pkg/proxy/kafka.go

```diff
@@ -204,41 +203,31 @@ func createKafkaRedirect(r *Redirect, conf kafkaConfiguration, endpointInfoRegis
 // canAccess determines if the kafka message req sent by identity is allowed to
 // be forwarded according to the rules configured on kafkaRedirect
 func (k *kafkaRedirect) canAccess(req *kafka.RequestMessage, srcIdentity identity.NumericIdentity) bool {
-	var id *identity.Identity
-
-	if srcIdentity != 0 {
-		id = cache.LookupIdentityByID(srcIdentity)
-		if id == nil {
-			log.WithFields(logrus.Fields{
-				logfields.Request:  req.String(),
-				logfields.Identity: srcIdentity,
-			}).Warn("Unable to resolve identity to labels")
-		}
-	}
-
 	scopedLog := log.WithFields(logrus.Fields{
-		logfields.Request:  req.String(),
-		logfields.Identity: id,
+		logfields.Request: req.String(),
+		"NumericIdentity": srcIdentity,
 	})
 
 	k.redirect.mutex.RLock()
-	rules := k.redirect.rules.GetRelevantRules(id)
+	rules := k.redirect.rules.GetRelevantRulesForKafka(srcIdentity)
 	k.redirect.mutex.RUnlock()
 
-	if rules.Kafka == nil {
+	if len(rules) == 0 {
 		flowdebug.Log(scopedLog, "No Kafka rules matching identity, rejecting")
 		return false
 	}
 
-	b, err := json.Marshal(rules.Kafka)
-	if err != nil {
-		flowdebug.Log(scopedLog, "Error marshalling kafka rules to apply")
-		return false
-	} else {
-		flowdebug.Log(scopedLog.WithField("rule", string(b)), "Applying rule")
+	if flowdebug.Enabled() {
```

Comment:
Add a comment that we are checking this explicitly to avoid Marshaling `rules` for no reason.